v2/src/ecs.ts

```diff
@@ -0,0 +1,145 @@
+                hostPort: 8125,
+                protocol: Protocol.TCP
+            }
+        ]
+
+    });
+    return datadogAgentContainer
+}
+
+export function addFluentBitRouter(options: EcsOptions, logGroup: LogGroup, ddSecret: EcsSecret): ContainerDefinition {
+
+    const fluentBitContainer = options.taskDefinition.addFirelensLogRouter(
+        `fluentBit-${options.taskDefinition.family}`, { // Adding a suffix to the container name to improve discoverability in infra monitoring
+        image: ContainerImage.fromRegistry("public.ecr.aws/aws-observability/aws-for-fluent-bit:init-latest"),
+        healthCheck: {
+            command: [
+                "echo",
+                '\'{"health": "check"}\'',
+                "|",
+                "nc",
+                "127.0.0.1",
+                "8877",
+                "||",
+                "exit 1",
+            ],
+            startPeriod: Duration.seconds(10),
+            retries: 5,
+            interval: Duration.seconds(5)
+        },
+        essential: true,
+        firelensConfig: {
+            type: FirelensLogRouterType.FLUENTBIT,
+        },
+        logging: LogDrivers.awsLogs({
+            streamPrefix: "fluentbit",
+            logGroup: logGroup
+        }),
+        secrets: { "DD_API_KEY": ddSecret }
+    })
+    return fluentBitContainer
+}
+
+export function updateTaskContainers(
+    ddAgentContainer: ContainerDefinition,
+    fluentBitContainer: ContainerDefinition,
+    options: EcsOptions
+) {
+    const cfnTaskDefinition = options.taskDefinition.node.defaultChild as CfnTaskDefinition
+    const taskDefinitionChildren = options.taskDefinition.node.children
+    for (const [index, child] of taskDefinitionChildren.entries()) {
+        if (child.constructor.name === ContainerDefinition.name) {
+            const container = child as ContainerDefinition
+            container.addContainerDependencies(
+                {
+                    container: ddAgentContainer,
+                    condition: ContainerDependencyCondition.HEALTHY
+                },
+                {
+                    container: fluentBitContainer,
+                    condition: ContainerDependencyCondition.HEALTHY
+                }
+            )
+            const datadogEnvVars = {
+                "DD_ENV": options.envName,
+                "DD_SERVICE": options.service,
+                "DD_VERSION": options.version || container.imageName,
+            }
+        
+            for (const [key, value] of Object.entries(datadogEnvVars)) {
+                container.addEnvironment(key, value)
+            }
+            cfnTaskDefinition.addPropertyOverride(
+                `ContainerDefinitions.${index}.LogConfiguration.Options`,
+                {
+                    "Name": "datadog",
+                    "Host": "http-intake.logs.datadoghq.com",
+                    "dd_service": options.service,
+                    "dd_source": container.containerName,
```

Comment:
It would be better if this could be set as a separate option. It's often useful to set this to `nodejs` or `ruby` to leverage the standard log processing pipelines

Reply:
Good call. Not sure the best way to define this. Might need to be passed as a separate mapping in a param in the method? Or maybe we could use [metadata](https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_cx-api.MetadataEntry.html)?